lib/ohai/mixin/alibaba_metadata.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+#
+# Author:: Tim Smith (<tsmith@chef.io>)
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "net/http" unless defined?(Net::HTTP)
+
+module Ohai
+  module Mixin
+    #
+    # This code parses the Alibaba Instance Metadata API to provide details
+    # of the running instance.
+    #
+    # As of 2021-02-07 there is only one API release so this removes the
+    # API parsing logic in the ec2 metadata mixin. If that changes we
+    # may want to merge these two codebases
+    #
```

Comment:
why split them, just to make someone else (probably you) merge them later? It's a versioned API, so it'll very likely get an additional version... why make that painful for us then?

Reply:
I need to actually remove that comment. It became clear pretty quickly that the approaches are very different and not something we could merge / make generic.